lib/email/receiver.rb

```diff
@@ -900,15 +901,39 @@ def add_attachments(raw, user_id, options = {})
             else
               raw << "\n\n#{attachment_markdown(upload)}\n\n"
             end
+          else
+            rejected_attachments << upload
+            raw << "\n\n#{I18n.t('emails.incoming.missing_attachment', filename: upload.original_filename)}\n\n"
           end
         ensure
           tmp&.close!
         end
       end
+      notify_about_rejected_attachment(rejected_attachments) if rejected_attachments.present? && !User.find(user_id).staged?
```

Comment:
Why not pass the `user` object instead of its `id` and use that in this method? This will prevent from doing a query to the database ;)